### PR TITLE
centos-ci: install vagrant from upstream instead of scl

### PR DESCRIPTION
**Pull request by obnoxxx**
vagrant in SCL is deprecated and does not install any more.
Because there's currently no good choice to work with centos7,
we are reverting to installing vagrant from upstream (hashicorp) RPMs.

Inspired by https://github.com/heketi/ci-tests/pull/9

Signed-off-by: Michael Adam <obnox@samba.org>

### CODE REVIEW

**Comment by obnoxxx** [comment]
@raghavendra-talur wrote:
> You need to add
>
> ```
>      black-list-target-branches:
>        - centos-ci
> ```
>
> to the job in https://github.com/gluster/centosci/blob/master/jobs/samba-integration.yml

No, it is actually *white* listed there: https://github.com/gluster/centosci/blob/master/jobs/samba-integration.yml#L48

We have the clue to restart the script if the PR changes it. And the "make test" target in the centos-ci branch will run it once more to actually call the test env from the master branch.


I will do a separate PR to do that PR-treatment earlier in the script, to be able to also test changes in the rpm install section...

**Comment by raghavendra-talur** [comment]
> @raghavendra-talur wrote:
> 
> > You need to add
> > ```
> >      black-list-target-branches:
> >        - centos-ci
> > ```
> > 
> > 
> > to the job in https://github.com/gluster/centosci/blob/master/jobs/samba-integration.yml
> 
> No, it is actually _white_ listed there: https://github.com/gluster/centosci/blob/master/jobs/samba-integration.yml#L48
> 

Ok, I thought you might want to disable the checks for the "ci" branch and keep it enabled only for the master. Based on https://docs.openstack.org/infra/jenkins-job-builder/triggers.html#triggers.github-pull-request

**Comment by obnoxxx** [comment]
I have created #48 to implement the splitting in phases so that the PR checking and restarting of the script is done early.

If we merge this first and rebase this afterwards, we should be able to start seeing the CI succeed already in this PR... ;-)

File: samba-integration-centos-ci-tests.sh [hunk 1]
```diff
@@ -17,26 +17,31 @@ SCRIPT_GIT_BRANCH="centos-ci"
...
-	sclo-vagrant1-vagrant \
-	sclo-vagrant1-vagrant-libvirt \
 	git \
 	make \
-	ansible
+	ansible \
+	libvirt \
+	libvirt-devel
+
+yum -y group install "Development Tools"
```
Comment (anoopcs9) [inline samba-integration-centos-ci-tests.sh:36]: Why do we need to install "Development Tools" group?

File: samba-integration-centos-ci-tests.sh [hunk 2]
```diff
@@ -17,26 +17,31 @@ SCRIPT_GIT_BRANCH="centos-ci"
...
+# enable additional sources for yum:
+# - epel for ansible
+yum -y install epel-release
 
 # Install additional packages
 #
-# note: adding sclo-vagrant1-vagrant explicitly seems to fix
-#   issues where libvirt fails to bring up the vm with errors like this:
-#   "Call to virDomainCreateWithFlags failed: the CPU is incompatible with host
-#    CPU: Host CPU does not provide required features: svm" (or vmx)
```
Comment (anoopcs9) [inline samba-integration-centos-ci-tests.sh:29]: I hope this is no longer an issue with upstream vagrant as it is running successfully with heketi CI job.

File: samba-integration-centos-ci-tests.sh [hunk 3]
```diff
@@ -17,26 +17,31 @@ SCRIPT_GIT_BRANCH="centos-ci"
...
 	git \
 	make \
-	ansible
+	ansible \
+	libvirt \
+	libvirt-devel
+
+yum -y group install "Development Tools"
+
+if ! rpm -q vagrant
```
Comment (anoopcs9) [inline samba-integration-centos-ci-tests.sh:38]: What are we anticipating here? Is it like _vagrant_ might get installed by default with CentOS box? Or as part of some dependency for other packages getting installed?

File: samba-integration-centos-ci-tests.sh (hunk 1 above)
Comment (obnoxxx) [reply samba-integration-centos-ci-tests.sh:36]: Needed in order to be able to do "vagrant plugin install"

File: samba-integration-centos-ci-tests.sh (hunk 3 above)
Comment (obnoxxx) [reply samba-integration-centos-ci-tests.sh:38]: yeah, I was wondering the same. Took it from the heketi patch. will remove here :-)

File: samba-integration-centos-ci-tests.sh (hunk 2 above)
Comment (obnoxxx) [reply samba-integration-centos-ci-tests.sh:29]: yes. hope this is consistently fixed now

File: samba-integration-centos-ci-tests.sh (hunk 1 above)
Comment (obnoxxx) [reply samba-integration-centos-ci-tests.sh:36]: adding a comment

**Comment by obnoxxx** [comment]
rebased on top of PR #48 - let's see how it goes :-)

File: samba-integration-centos-ci-tests.sh (hunk 3 above)
Comment (obnoxxx) [reply samba-integration-centos-ci-tests.sh]: Hmmm, it now failed with:

```
19:27:36 + yum -y install https://releases.hashicorp.com/vagrant/2.2.7/vagrant_2.2.7_x86_64.rpm
19:27:36 Loaded plugins: fastestmirror
19:27:37 Examining /var/tmp/yum-root-5IF4Zr/vagrant_2.2.7_x86_64.rpm: 1:vagrant-2.2.7-1.x86_64
19:27:37 /var/tmp/yum-root-5IF4Zr/vagrant_2.2.7_x86_64.rpm: does not update installed package.
19:27:37 Error: Nothing to do
19:27:37 make: *** [test] Error 1
```

I think it is related to us restarting the script w/o the PR . Checking it and will update...

**Comment by obnoxxx** [comment]
So it failed with:

```
19:27:36 + yum -y install https://releases.hashicorp.com/vagrant/2.2.7/vagrant_2.2.7_x86_64.rpm
19:27:36 Loaded plugins: fastestmirror
19:27:37 Examining /var/tmp/yum-root-5IF4Zr/vagrant_2.2.7_x86_64.rpm: 1:vagrant-2.2.7-1.x86_64
19:27:37 /var/tmp/yum-root-5IF4Zr/vagrant_2.2.7_x86_64.rpm: does not update installed package.
19:27:37 Error: Nothing to do
19:27:37 make: *** [test] Error 1
```

I think it is related to us restarting the script w/o the PR.
It is a weird behavior of `yum install` which fails if the package is already installed and at the desired version. So I used a hack of `yum install ...  || yum update ...` to work around it. it's the most elegant solution I found.

**Comment by obnoxxx** [comment]
Fingers crossed now ... :-D

**Pull request merged**